### PR TITLE
refactor(proteus): Use constructor instead of ClassUtil (BREAKING) (WEBAPP-6789)

### PR DESCRIPTION
**Pull request by ffflorian**
This PR will
- remove the `ClassUtil` and use built-in constructors
- remove the switch case for decoding and just walk through the object instead
- convert snake_case to camelCase where possible without breaking things

**BREAKING CHANGES:**
- `Proteus.keys.PreKeyBundle.new()` ➡️ `new Proteus.keys.PreKeyBundle()`

## Pull Request Checklist

- [x] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/proteus/src/main/derived/DerivedSecrets.ts [hunk 1]
```diff
@@ -29,7 +27,7 @@ export class DerivedSecrets {
   mac_key: MacKey;
 
   constructor() {
-    this.cipher_key = new CipherKey();
+    this.cipher_key = new CipherKey(new Uint8Array([]));
```
Comment (bennyn) [inline packages/proteus/src/main/derived/DerivedSecrets.ts:30]: Can `Uint8Array` become a mandatory property of the `constructor`?

File: packages/proteus/spec/derived/CipherKeySpec.ts [hunk 2]
```diff
@@ -96,8 +96,8 @@ describe('CipherKey sanity checks (IETF ChaCha20 test vectors)', () => {
       };
     };
 
-    ietf_vectors.forEach(vector => {
-      const result = decrypt_cipher_text(vector);
+    ietfVctors.forEach(vector => {
```
Comment (bennyn) [inline packages/proteus/spec/derived/CipherKeySpec.ts:99]:
```suggestion
    ietfVectors.forEach(vector => {
```

File: packages/proteus/spec/derived/CipherKeySpec.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ beforeAll(async () => {
 });
 
 describe('CipherKey sanity checks (IETF ChaCha20 test vectors)', () => {
-  const ietf_vectors: Vector[] = [
+  const ietfVctors: Vector[] = [
```
Comment (bennyn) [inline packages/proteus/spec/derived/CipherKeySpec.ts:34]:
```suggestion
  const ietfVectors: Vector[] = [
```

File: packages/proteus/src/main/derived/DerivedSecrets.ts (hunk 1 above)
Comment (ffflorian) [reply packages/proteus/src/main/derived/DerivedSecrets.ts]: Yes :)

**Pull request merged**